### PR TITLE
functional tests - part 1

### PR DESCRIPTION
**Pull request by emdobrin**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java [hunk 1]
```diff
@@ -0,0 +1,646 @@
+/*
```
Comment (addb) [inline code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:1]: 2023?

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+			signalRequest,
+			new MockConnection(
+				200,
+				"{\"uuid\":\"19994521975870785742420741570375407533\", \"stuff\":[{\"cv\":\"cv\",\"cn\":\"cn\"}]}"
+			)
+		);
+		testableNetworkService.setExpectedNetworkRequest(signalRequest, 1);
+
+		// preset the config and shared Preferences
+		//mockUUIDInPersistence();
```
Comment (addb) [inline code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:400]: remove this line

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+		registerExtensions(config);
```
Comment (addb) [inline code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:401]: MobileCore.updateConfiguration(config);

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+		assertEquals(0, testableNetworkService.getReceivedNetworkRequestsMatching(signalRequest).size());
+
+		// Privacy changes to opt in
+		MobileCore.setPrivacyStatus(MobilePrivacyStatus.OPT_IN);
+		testableNetworkService.assertNetworkRequestCount();
+		List<TestableNetworkRequest> requests = testableNetworkService.getReceivedNetworkRequestsMatching(
+			signalRequest
+		);
+		assertEquals(2, requests.size());
+		assertTrue(requests.get(0).getUrl().contains("https://server/event?"));
```
Comment (addb) [inline code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:474]: should we verify the second request as well?

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+
+		// dispatch signal
+		Audience.signalWithData(signalData, null);
+
+		// verify that no network call is made
+		assertEquals(0, testableNetworkService.getReceivedNetworkRequestsMatching(signalRequest).size());
+
+		// configure
+		config.put("global.privacy", "optedin");
+		MobileCore.updateConfiguration(config);
```
Comment (addb) [inline code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:360]: should we use `MobileCore.setPrivacyStatus(MobilePrivacyStatus.OPT_IN);` in place of updateConfig?

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+	}
+
+	@Test
+	public void testLifecycleEvent_AndSubmitSignal_AndConfigUpdate_thenShouldSendTwoRequestsInCorrectOrder()
+		throws InterruptedException {
+		// setup
+		TestableNetworkRequest signalRequest = new TestableNetworkRequest("https://server/event", HttpMethod.GET);
+		testableNetworkService.setExpectedNetworkRequest(signalRequest, 2);
+
+		config.put("global.privacy", "optunknown");
```
Comment (addb) [inline code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:346]: should we use `MobileCore.setPrivacyStatus(MobilePrivacyStatus.OPT_UNKNOWN);` in place of updateConfig?

File: code/audience/build.gradle [hunk 7]
```diff
@@ -235,12 +235,7 @@ dependencies {
     testImplementation "org.mockito:mockito-inline:${rootProject.ext.mockitoCoreVersion}"
```
Comment (addb) [inline code/audience/build.gradle:235]: implementation 'androidx.annotation:annotation:1.0.0'

**Comment by addb** [comment]
Have left few comments. Otherwise looks good!

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java (hunk 4 above)
Comment (emdobrin) [reply code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:472]: verified below on `requests.get(1)`

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java (hunk 3 above)
Comment (emdobrin) [reply code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:399]: registerExtensions with config calls updateConfiguration underneath for quick setup

File: code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java (hunk 6 above)
Comment (emdobrin) [reply code/audience/src/androidTest/java/com/adobe/marketing/mobile/audience/AudienceFunctionalTests.java:346]: again, this time just overwriting config with privacy unknown and setting the config on registration

**Review by addb — APPROVED** [review]
Looks good!

**Pull request merged**